test/metric_value_benchmark/collectd_test.go

```diff
@@ -0,0 +1,84 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+//go:build linux && integration
+// +build linux,integration
+
+package metric_value_benchmark
+
+import (
+	"time"
+
+	"github.com/aws/amazon-cloudwatch-agent-test/test/metric"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/status"
+)
+
+type CollectDTestRunner struct {
+}
+
+var _ ITestRunner = (*CollectDTestRunner)(nil)
+
+func (t *CollectDTestRunner) validate() status.TestGroupResult {
+	metricsToFetch := t.getMeasuredMetrics()
+	testResults := make([]status.TestResult, len(metricsToFetch))
+	for i, metricName := range metricsToFetch {
+		testResults[i] = validateCollectDMetric(metricName)
+	}
+
+	return status.TestGroupResult{
+		Name:        t.getTestName(),
+		TestResults: testResults,
+	}
+}
+
+func (t *CollectDTestRunner) getTestName() string {
+	return "CollectD"
+}
+
+func (t *CollectDTestRunner) getAgentConfigFileName() string {
+	return "collectd_config.json"
+}
+
+func (t *CollectDTestRunner) getAgentRunDuration() time.Duration {
+	return minimumAgentRuntime
+}
+
+func (t *CollectDTestRunner) getExtraCommands() []string {
+	return []string{
+		"sudo mkdir -p /etc/collectd",
+		"sudo chown `whoami` /etc/collectd",
+		"echo 'user: secret' > /etc/collectd/auth_file",
+		fmt."sudo collectd -C ./extra_configs/collectd.conf",
+	}
+}
```

Comment:
What is this for? Should this be baked into the AMI

Reply:
So the [collectd](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch-Agent-custom-metrics-collectd.html) has the option to send the data with encryption. Therefore, `secret` is used to verify signatures and to decrypt encrypted network packets. IMO, it should not be baked into the AMI since it needs to be configured in both [collectd.conf](https://github.com/aws/amazon-cloudwatch-agent-test/pull/13/files#diff-0c4c6e0c9498578e2b6b38316e2093258e2c060b84f7e83c2bf170e1ac8f21e6R13-R14)

Reply:
Do we need to add this for testing?

Reply:
Not entirely if we only consider the `metric value benchmark` However, I'm not seeing the harm of adding it too.

Reply:
What is the value for adding this? This adds complexity to our testing

Reply:
Would be simulating the flow for customers but after discussing in OH, I have deleted it !